R2API.Skins/SkinVFX.cs

```diff
@@ -0,0 +1,229 @@
+/// </summary>
+public static partial class SkinVFX {
+    private static List<SkinVFXInfo> skinVFXInfos = new List<SkinVFXInfo>();
+    private static bool hooksSet = false;
+    private const uint BaseIdentifier = 24000; // arbitrary, but we shouldn't hit 24,000 unique items for a substantial amount of time.
+    private static uint currentIdentifier = BaseIdentifier; 
+    private static uint nextIdentifier => currentIdentifier++;
+
+    /// <summary>
+    /// Called when a skin-specific effect is ready to be modified.
+    /// </summary>
+    /// <param name="spawnedEffect">The effect that was spawned.</param>
+    public delegate void OnEffectSpawnedDelegate(GameObject spawnedEffect);
+
+    private static bool hasCatalogInitOccured = false;
+
+
+    internal static void SetHooks() {
+        if (hooksSet) {
+            return;
+        }
+
+        hooksSet = true;
+
+        On.RoR2.EffectComponent.Start += ApplyModifier;
+        On.RoR2.EffectManager.SpawnEffect_GameObject_EffectData_bool += ApplyReplacement;
+        IL.RoR2.BulletAttack.FireSingle += ModifyBulletAttack;
+        IL.RoR2.EffectManager.SimpleMuzzleFlash += ModifyMuzzleFlash;
+        On.EntityStates.BasicMeleeAttack.BeginMeleeAttackEffect += ModifyGenericMelee;
+
+        IL.RoR2.Orbs.GenericDamageOrb.Begin += ModifyGenericOrb;
+    }
+
+    internal static void UnsetHooks() {
+        hooksSet = false;
+        On.RoR2.EffectComponent.Start -= ApplyModifier;
+        On.RoR2.EffectManager.SpawnEffect_GameObject_EffectData_bool -= ApplyReplacement;
+        IL.RoR2.BulletAttack.FireSingle -= ModifyBulletAttack;
+        IL.RoR2.EffectManager.SimpleMuzzleFlash -= ModifyMuzzleFlash;
+        On.EntityStates.BasicMeleeAttack.BeginMeleeAttackEffect -= ModifyGenericMelee;
+
+        IL.RoR2.Orbs.GenericDamageOrb.Begin -= ModifyGenericOrb;
+    }
+
+    [SystemInitializer(typeof(EffectCatalog))]
+    private static void FindEffectIndexes() {
+        hasCatalogInitOccured = true;
+
+        for (int i = 0; i < skinVFXInfos.Count; i++) {
+            SkinVFXInfo skinVFXInfo = skinVFXInfos[i];
+
+            if (skinVFXInfo.EffectPrefab) {
+                skinVFXInfo.TargetEffect = EffectCatalog.FindEffectIndexFromPrefab(skinVFXInfo.EffectPrefab);
+                continue;
+            }
+
+            if (!String.IsNullOrEmpty(skinVFXInfo.EffectString)) {
+                EffectDef def = EffectCatalog.entries.FirstOrDefault(effectDef => effectDef.prefabName == skinVFXInfo.EffectString);
+
+                if (def == null) {
+                    SkinsPlugin.Logger.LogError($"Failed to find effect {skinVFXInfo.EffectString} for SkinVFXInfo!");
+                    continue;
+                }
+
+                skinVFXInfo.TargetEffect = def.index;
+            }
+        }
+    }
+
+    private static SkinVFXInfo FindSkinVFXInfo(uint identifier) {
+        return skinVFXInfos.FirstOrDefault(skinVFXInfo => skinVFXInfo.Identifier == identifier);
```

Comment:
In that case,I recomment 
```suggestion
        return skinVFXInfos[identifier - BaseIdentifier];
```
So that the cost of repeated lookups is lessened.